### ISSUE TITLE
Error when trying to fetch a GH-hosted library: `fatal: unable to access URI`

### ISSUE THREAD

**Issue by perezzini** [pypa/pipenv#5859]
Hello,

Using latest pipenv version (https://pypi.org/project/pipenv/2023.8.22/), and trying to install a package from outside PYPI, such as GH:
```shell
ext-lib-example = {editable = true, ref = "master", git = "https://${TOKEN}@github.com/org/ext-lib-example.git"}
```
the following error occurs:
```shell
[pipenv.exceptions.InstallError]: Collecting ext-lib-example@ git+https://***@347c4f2de42c03221abe33ee51878b5f11bf4874 (from -r /tmp/pipenv-ru2z44tb-requirements/pipenv-a0a4ustz-reqs.txt (line 1))
[pipenv.exceptions.InstallError]:   Cloning https://****@347c4f2de42c03221abe33ee51878b5f11bf4874 to /tmp/pip-install-319b9osz/ext-lib-example_327d9e97969d4bed8b03a8[66]
[pipenv.exceptions.InstallError]: Running command git clone --filter=blob:none --quiet 'https://****@347c4f2de42c03221abe33ee51878b5f11bf4874' /tmp/pip-install-319b9osz/ext-lib-example_327d9e97969d4bed8b03a866e6366b2c
[pipenv.exceptions.InstallError]:   fatal: unable to access 'https://347c4f2de42c03221abe33ee51878b5f11bf4874/': Could not resolve host: 347c4f2de42c03221abe33ee51878b5f11bf4874
[pipenv.exceptions.InstallError]:   error: subprocess-exited-with-error
[pipenv.exceptions.InstallError]:   
[pipenv.exceptions.InstallError]:   × git clone --filter=blob:none --quiet 'https://****@347c4f2de42c03221abe33ee51878b5f11bf4874' /tmp/pip-install-319b9osz/ext-lib-example_327d9e979[69]
[pipenv.exceptions.InstallError]:   │ exit code: 128
[pipenv.exceptions.InstallError]:   ╰─> See above for output.
```
Had to downgrade to https://pypi.org/project/pipenv/2023.7.23/ version.

**Comment by dhendry** [comment]
I too am just trying to debug this error.

It appears that pipenv is dropping the `github.com/[repo name]` part of the path

**Comment by matteius** [comment]
I believe this one solved by https://github.com/pypa/pipenv/pull/5856  which just got merged into main -- if you could check against latest main that would be helpful.

I plan to cut a release yet this evening.

**Comment by dmelo** [comment]
I'm having the same problem with 2023.8.22, but I had to downgrade only to 2023.8.20 . The error started on 2023.8.21 , at least in my case.

**Comment by perezzini** [comment]
> I believe this one solved by #5856 which just got merged into main -- if you could check against latest main that would be helpful.
> 
> I plan to cut a release yet this evening.

@matteius using latest main version appears to work fine.

Would wait for a release, then. Thanks!

**Comment by matteius** [comment]
`2023.8.23` has been cut for the vcs issues.
